crates/agent/src/api/snapshot.rs

```diff
@@ -451,7 +470,8 @@ async fn try_fetch(pg_pool: &sqlx::PgPool) -> anyhow::Result<Snapshot> {
             d.data_plane_name,
             d.data_plane_fqdn,
             false AS "is_default!: bool",
-            d.hmac_keys,
+            '{}'::text[] as "hmac_keys!",
```

Comment:
You should fetch both, and then use `encrypted_hmac_keys` only if `hmac_keys` is non-empty.

That lets us release this, and then update pulumi to introduce encrypted keys everywhere, and then remove the non-encrypted ones to cut over and verify nothing breaks, with very easy rollback points along the way.